hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -360,17 +361,18 @@ commit' ::
   HeadId ->
   -- | Spendable 'UTxO'
   UTxO ->
-  -- | 'UTxO' to commit, along with witnesses to spend them.
-  UTxO' (TxOut CtxUTxO, Witness WitCtxTxIn) ->
+  -- | 'UTxO' to commit
+  UTxO ->
+  Tx ->
   Either (PostTxError Tx) Tx
-commit' ctx headId spendableUTxO utxoToCommit = do
+commit' ctx headId spendableUTxO utxoToCommit blueprintTx = do
   pid <- headIdToPolicyId headId ?> InvalidHeadId{headId}
   (i, o) <- ownInitial pid ?> CannotFindOwnInitial{knownUTxO = spendableUTxO}
-  let utxo = fst <$> utxoToCommit
-  rejectByronAddress utxo
-  rejectReferenceScripts utxo
-  rejectMoreThanMainnetLimit networkId utxo
-  pure $ commitTx networkId scriptRegistry headId ownParty utxoToCommit (i, o, vkh)
+  -- FIXME: not all outputs in utxoToCommit are necessarily commited, they could also be reference inputs to the blueprintTx
```

Comment:
are we resolving this in a separate PR or shall we solve it in this one?

Reply:
good one. @ch1bo we said we will accept the potential user provided reference scripts right?